pkg/compact/compact.go

```diff
@@ -1201,7 +1235,13 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			go func() {
 				defer wg.Done()
 				for g := range groupChan {
-					shouldRerunGroup, _, err := g.Compact(workCtx, c.compactDir, c.planner, c.comp)
+					var err error
+					var shouldRerunGroup bool
+					// done in a child span derived from the parent span in ctx
+					tracing.DoInSpan(ctx, "group_compaction", func(workCtx context.Context) {
+						shouldRerunGroup, _, err = g.Compact(workCtx, c.compactDir, c.planner, c.comp)
+					}) //not adding group key for group_compaction since it is already added to "compaction" parent span
```

Comment:
To make things simpler, we can remove the span here and instrument in https://github.com/metonymic-smokey/thanos/blob/tracing/pkg/compact/compact.go#L769 instead. Should have the same effect.
Also let's add the group key here